### PR TITLE
error -> anyerror coercion, comptime zirErrorName

### PR DESCRIPTION
**Pull request by g-w1**

### CODE REVIEW

File: src/Sema.zig [hunk 1]
```diff
@@ -7090,9 +7090,28 @@ fn zirEmbedFile(sema: *Sema, block: *Scope.Block, inst: Zir.Inst.Index) CompileE
 }
 
 fn zirErrorName(sema: *Sema, block: *Scope.Block, inst: Zir.Inst.Index) CompileError!Air.Inst.Ref {
+    const tracy = trace(@src());
+    defer tracy.end();
+
     const inst_data = sema.code.instructions.items(.data)[inst].un_node;
     const src = inst_data.src();
-    return sema.mod.fail(&block.base, src, "TODO: Sema.zirErrorName", .{});
+    _ = src;
```
Comment (miscco) [inline src/Sema.zig:7098]: Out of curiosity what is the purpose of this assignment?

`src` is always used in the if below

Comment (g-w1) [reply src/Sema.zig:7098]: Thanks for the review, I am cleaning up someone else's code, will remove.

File: src/type.zig [hunk 2]
```diff
@@ -1145,6 +1145,106 @@ pub const Type = extern union {
         }
     }
 
+    pub fn isGlobalErrorSet(ty: Type) bool {
```
Comment (andrewrk) [inline src/type.zig:1148]: The logic for this is unfortunately not this simple. A Type could be a `error_set_inferred` in which case it *might* be the global error set. Determining whether an inferred error set is the global error set requires "resolving" the inferred error set, which could cause compile errors to be emitted. It means we cannot implement this function here.

**Review by andrewrk — CHANGES_REQUESTED** [review]

File: src/type.zig (hunk 2 above)
Comment (g-w1) [reply src/type.zig:1148]: I have stubbed out isGlobalErrorSet and put it in Sema.zig for now, as it can return an error if the resolution fails.

**Pull request closed without merging**